### PR TITLE
C++: Autoformat QL code in Architecture and Best Practices

### PR DESCRIPTION
**Pull request by jbj**
This is my attempt to get started autoformatting all our C++ QL code. This PR formats the code in `Architecture` and `Best Practices`. Those directories don't get changed very often, so conflicts should be unlikely.

I don't think the output of the auto-formatter is perfect, but I think it's overall better than the input. In addition, I think it has great value to unify our formatting.

These changes were made with the following commands:

```
./build -j4 target/jars/qlformat
find ql/cpp/ql/src/{Architecture,Best\ Practices} -name "*.ql*" -print0 | xargs -0 target/jars/qlformat --input
```

### CODE REVIEW

File: cpp/ql/src/Architecture/General Top-Level Information/GeneralStatistics.ql [hunk 1]
```diff
@@ -5,31 +5,57 @@
...
+              c.fromSource() and toSum = c.getMetrics().getEfferentSourceCoupling()
+            |
+              toSum
+            ) /
+          sum(Class c, int toSum |
+            c.fromSource() and toSum = c.getMetrics().getEfferentCoupling()
+          |
+            toSum
+          )
+      ).toString() + "%"
```
Comment (geoffw0) [inline cpp/ql/src/Architecture/General Top-Level Information/GeneralStatistics.ql]: It's not a big deal, but I'm not especially keen on what's happened with the arithmetic here.

File: cpp/ql/src/Best Practices/Magic Constants/MagicConstants.qll [hunk 2]
```diff
@@ -2,62 +2,208 @@ import cpp
...
+  s = "0x00" or
+  s = "10" or
+  s = "100" or
+  s = "1000" or
+  s = "10000" or
+  s = "100000" or
+  s = "1000000" or
+  s = "10000000" or
+  s = "100000000" or
+  s = "1000000000"
```
Comment (geoffw0) [inline cpp/ql/src/Best Practices/Magic Constants/MagicConstants.qll]: In this case I much prefer the original.  We've lost a bit of information and a lot of readability.  Comments might help???

File: cpp/ql/src/Architecture/General Top-Level Information/GeneralStatistics.ql (hunk 1 above)
Comment (jbj) [reply cpp/ql/src/Architecture/General Top-Level Information/GeneralStatistics.ql]: Fixed by rewriting the expressions to fit within the 100-colum limit. That should be correct according to https://semmle.slack.com/archives/C20TR4EAZ/p1555058003000800.

File: cpp/ql/src/Best Practices/Magic Constants/MagicConstants.qll (hunk 2 above)
Comment (jbj) [reply cpp/ql/src/Best Practices/Magic Constants/MagicConstants.qll]: Comments added.

**Pull request merged**